### DISCUSSION TITLE
Starting out of the box with Mongo

### DISCUSSION THREAD

**Discussion by KisaPs** [Aguafrommars/TheIdServer#1220] [Q&A]
**### What basic settings were not taken into account by me for the first launch? According to the documentation, I have not yet seen the required settings, I am not interested in the exact configuration of the server for review.**

Considering TheIdServer, I installed the server template (VS), and based on the documentation, I made some settings. Changed the database to MongoDB, connectionstring. At the first launch, SeedData fills the database. An error crashes

`
Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Collections.Generic.KeyNotFoundException: The given key 'persistAuthorization' was not present in the dictionary.
   at System.Collections.Generic.Dictionary2.get_Item(TKey key)
   at NSwag.AspNetCore.SwaggerUiSettings.get_PersistAuthorization()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
   at System.Reflection.PropertyInfo.GetValue(Object obj)
   at Microsoft.Extensions.Configuration.BindingPoint.get_Value()
   at Microsoft.Extensions.Configuration.ConfigurationBinder.BindProperty(PropertyInfo property, Object instance, IConfiguration config, BinderOptions options)
   at Microsoft.Extensions.Configuration.ConfigurationBinder.BindProperties(Object instance, IConfiguration configuration, BinderOptions options)
   at Microsoft.Extensions.Configuration.ConfigurationBinder.BindInstance(Type type, BindingPoint bindingPoint, IConfiguration config, BinderOptions options, Boolean isParentCollection)
   at Microsoft.Extensions.Configuration.ConfigurationBinder.Get(IConfiguration configuration, Type type, Action 1 configureOptions)
   at Microsoft.Extensions.Configuration.ConfigurationBinder.Get[T](IConfiguration configuration, Action 1 configureOptions)
   at Microsoft.AspNetCore.Builder.ApplicationBuilderExtensions.<>c__DisplayClass0_0.<UseTheIdServer>b__5(SwaggerUiSettings options) in D:\source\TheIdServer\src\TheIdServer\Extensions\ApplicationBuilderExtensions.cs:line 97
   at Microsoft.AspNetCore.Builder.NSwagApplicationBuilderExtensions.UseSwaggerUi(IApplicationBuilder app, Action 1 configure)
   at Microsoft.AspNetCore.Builder.ApplicationBuilderExtensions.<>c__DisplayClass0_0.<UseTheIdServer>b__1(IApplicationBuilder child) in D:\source\TheIdServer\src\TheIdServer\Extensions\ApplicationBuilderExtensions.cs:line 94
   at Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseIdentityServerAdminApi(IApplicationBuilder builder, String basePath, Action 1 configure)
   at Microsoft.AspNetCore.Builder.ApplicationBuilderExtensions.UseTheIdServer(IApplicationBuilder app, IWebHostEnvironment environment, IConfiguration configuration) in D:\source\TheIdServer\src\TheIdServer\Extensions\ApplicationBuilderExtensions.cs:line 88
   at Program.<Main>$(String[] args) in D:\source\TheIdServer\src\TheIdServer\Program.cs:line 46`

Then I disable Swagger via appsettings "EnableOpenApiDoc": false. The application starts with an error 

`[10:04:30 ERR] An exception was thrown while deserializing the token.
Microsoft.AspNetCore.Antiforgery.AntiforgeryValidationException: The antiforgery token could not be decrypted.
 ---> System.Security.Cryptography.CryptographicException: The key {2f308a20-c4e0-4d3d-889a-f87c53b3920c} was not found in the key ring. For more information go to https://aka.ms/aspnet/dataprotectionwarning
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.UnprotectCore(Byte[] protectedData, Boolean allowOperationsOnRevokedKeys, UnprotectStatus& status)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.Unprotect(Byte[] protectedData)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)
[10:04:30 ERR] An exception was thrown while deserializing the token.
Microsoft.AspNetCore.Antiforgery.AntiforgeryValidationException: The antiforgery token could not be decrypted.
 ---> System.Security.Cryptography.CryptographicException: The key {2f308a20-c4e0-4d3d-889a-f87c53b3920c} was not found in the key ring. For more information go to https://aka.ms/aspnet/dataprotectionwarning
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.UnprotectCore(Byte[] protectedData, Boolean allowOperationsOnRevokedKeys, UnprotectStatus& status)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.Unprotect(Byte[] protectedData)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext).`

Then I add
`"DataProtectionOptions": {
"StorageKind": "MongoDB",
"KeyManagementOptions": {
"AutoGenerateKeys": true,
"NewKeyLifetime": "90.00:00:00"
},
"AuthenticatedEncryptorConfiguration": {
"EncryptionAlgorithm": "AES_256_CBC",
"ValidationAlgorithm": "HMACSHA256"
}
}.`

I get crit: 
` Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: '<' is an invalid start of a value. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
System.Text.Json.JsonException: '<' is an invalid start of a value. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: '<' is an invalid start of a value. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan1 bytes)
   at System.Text.Json.Utf8JsonReader.ConsumeValue(Byte marker)
   at System.Text.Json.Utf8JsonReader.ReadFirstToken(Byte first)
   at System.Text.Json.Utf8JsonReader.ReadSingleSegment()
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter1[[Aguacongas.IdentityServer.Store.PageResponse1[[Aguacongas.IdentityServer.Store.Entity.Culture, Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]], Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]].ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter1[[Aguacongas.IdentityServer.Store.PageResponse1[[Aguacongas.IdentityServer.Store.Entity.Culture, Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]], Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]].ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo1[[Aguacongas.IdentityServer.Store.PageResponse1[[Aguacongas.IdentityServer.Store.Entity.Culture, Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]], Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]].Deserialize(Utf8JsonReader& reader, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadFromSpan[PageResponse1](ReadOnlySpan1 utf8Json, JsonTypeInfo1 jsonTypeInfo, Nullable1 actualByteCount)
   at System.Text.Json.JsonSerializer.ReadFromSpan[PageResponse1](ReadOnlySpan1 json, JsonTypeInfo1 jsonTypeInfo)
   at System.Text.Json.JsonSerializer.Deserialize[PageResponse1](String json, JsonSerializerOptions options)
   at Aguacongas.IdentityServer.Admin.Http.Store.HttpStoreBase1.<DeserializeResponse>d__17 1[[Aguacongas.IdentityServer.Store.Entity.Culture, Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null],[Aguacongas.IdentityServer.Store.PageResponse1[[Aguacongas.IdentityServer.Store.Entity.Culture, Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]], Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
   at Aguacongas.IdentityServer.Admin.Http.Store.HttpStoreBase1.<GetAsync>d__14[[Aguacongas.IdentityServer.Store.Entity.Culture, Aguacongas.IdentityServer.Store, Version=8.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
   at Aguacongas.TheIdServer.BlazorApp.Infrastructure.Services.StringLocalizer.GetSupportedCulturesAsync()
   at Aguacongas.TheIdServer.BlazorApp.Components.LoginDisplay.OnInitializedAsync()
   at Microsoft.AspNetCore.Components.ComponentBase.RunInitAndSetParametersAsync()
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)`

When trying to navigate through the Web Admin pages, I see that LazyLoad shows
` Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Error: Aguacongas.TheIdServer.BlazorApp.Pages.Users.dll must be marked with 'BlazorWebAssemblyLazyLoad' item group in your project file to allow lazy-loading.
Error: Aguacongas.TheIdServer.BlazorApp.Pages.Users.dll must be marked with 'BlazorWebAssemblyLazyLoad' item group in your project file to allow lazy-loading.`

**Comment by aguacongas** [comment]
Can you share your appsettings.json or send it to by eMail ?

**Reply by KisaPs** [reply]
### The main problem is LazyLoad and CSP
**appsettings.json** - Barely touched it

```json
{
  "ConnectionStrings": {
    "DefaultConnection": "mongodb://localhost:27017/TIS"
  },
  "SiteOptions": {
    "Name": "TheIdServer"
  },
  "IdentityOptions": {
    "User": {
      "AllowedUserNameCharacters": "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+/\\ "
    },
    "SignIn": {
      "RequireConfirmedAccount": true
    }
  },
  "CertificateAuthenticationOptions": {
    "AllowedCertificateTypes": "All",
    "ValidateCertificateUse": false,
    "ValidateValidityPeriod": false,
    "RevocationMode": "NoCheck"
  },
  "IdentityServerOptions": {
    "MutualTls": {
      "Enabled": true,
      "PEMHeader": "ssl-client-cert"
    },
    "EnableServerSideSession": true,
    "ServerSideSessions": {
      "UserDisplayNameClaimType": "name",
      "RemoveExpiredSessions": true,
      "ExpiredSessionsTriggerBackchannelLogout": true,
      "RemoveExpiredSessionsFrequency": "00:10:00",
      "RemoveExpiredSessionsBatchSize": 100
    },
    "CustomEntriesOfStringArray": {
      "token_endpoint_auth_signing_alg_values_supported": [
        "RS256",
        "ES256",
        "ES384",
        "ES512",
        "PS256",
        "PS384",
        "PS512",
        "RS384",
        "RS512"
      ]
    }
  },
  "AccountOptions": {
    "ShowLogoutPrompt": false,
    "AutomaticRedirectAfterSignOut": true
  },
  "IdentityServer": {
    "Key": {
      "Type": "KeysRotation",
      "StorageKind": "MongoDb",
      "AdditionalSigningKeyType": {
        "RS384": {
          "SigningAlgorithm": "RS384"
        },
        "RS512": {
          "SigningAlgorithm": "RS512"
        },
        "PS256": {
          "SigningAlgorithm": "PS256"
        },
        "PS384": {
          "SigningAlgorithm": "PS384"
        },
        "PS512": {
          "SigningAlgorithm": "PS512"
        },
        "ES256": {
          "SigningAlgorithm": "ES256"
        },
        "ES384": {
          "SigningAlgorithm": "ES384"
        },
        "ES512": {
          "SigningAlgorithm": "ES512"
        }
      }
    }
  },
  "DataProtectionOptions": {
    "StorageKind": "MongoDb",
    "StorageConnectionString": "mongodb://localhost:27017/TIS4",
    "ApplicationName": "TheIdServer",
    "KeyManagementOptions": {
      "AutoGenerateKeys": true,
      "NewKeyLifetime": "90.00:00:00"
    }
  },
  "DisableTokenCleanup": false,
  "TokenCleanupInterval": "00:05:00",
  "Proxy": false,
  "Migrate": true,
  "DbType": "MongoDb",
  "Seed": true,
  "AuthenticatorIssuer": "TheIdServer",
  "EnableOpenApiDoc": false,
  "CorsAllowedOrigin": [
    "http://localhost:5001",
    "http://localhost:5002"
  ],
  "SwaggerUiSettings": {
    "Path": "/api/swagger",
    "OAuth2Client": {
      "ClientId": "theidserver-swagger",
      "ClientSecret": "client-secret",
      "AppName": "TheIdServer Swagger UI",
      "UsePkceWithAuthorizationCodeGrant": false
    },
    "WithCredentials": false
  },
  "ApiAuthentication": {
    "Authority": "https://localhost:5443",
    "RequireHttpsMetadata": false,
    "ApiName": "theidserveradminapi",
    "ApiSecret": "5b556f7c-b3bc-4b5b-85ab-45eed0cb962d",
    "EnableCaching": true,
    "CacheDuration": "0:10:0"
  },
  "PrivateServerAuthentication": {
    "Authority": "https://localhost:5443",
    "ApiUrl": "https://localhost:5443/api",
    "HeathUrl": "https://localhost:5443/healthz",
    "ClientId": "public-server",
    "ClientSecret": "84137599-13d6-469c-9376-9e372dd2c1bd",
    "Scope": "theidserveradminapi",
    "HttpClientName": "is4"
  },
  "EmailApiAuthentication": {
    "Authority": "https://localhost:5443",
    "ApiUrl": "https://localhost:5443/api/email",
    "ClientId": "public-server",
    "ClientSecret": "84137599-13d6-469c-9376-9e372dd2c1bd",
    "Scope": "theidserveradminapi",
    "HttpClientName": "email"
  },
  "BackchannelAuthenticationUserNotificationServiceOptions": {
    "Authority": "https://localhost:5443",
    "ApiUrl": "https://localhost:5443/api/email",
    "ClientId": "public-server",
    "ClientSecret": "84137599-13d6-469c-9376-9e372dd2c1bd",
    "Scope": "theidserveradminapi",
    "HttpClientName": "ciba"
  },
  "DynamicConfigurationOptions": {
    "ProviderType": "Aguacongas.DynamicConfiguration.Redis.RedisConfigurationProvider, Aguacongas.DynamicConfiguration.Redis"
  },
  "DynamicClientRegistrationOptions": {
    "Protected": false,
    "AllowedContacts": [
      {
        "Contact": "certification@oidf.org",
        "AllowedHosts": [
          "www.certification.openid.net"
        ]
      }
    ]
  },
  "RedisConfigurationOptions": {
    "ConnectionString": "localhost",
    "HashKey": "Aguacongas.TheIdServer.Duende",
    "Channel": "Aguacongas.TheIdServer.Duende.Channel"
  },
  "SignalR": {
    "HubOptions": {
      "EnableDetailedErrors": true
    },
    "UseMessagePack": true,
    "RedisOptions": {
      "Configuration": {
        "ChannelPrefix": "TheIdServer"
      }
    }
  },
  "Serilog": {
    "LevelSwitches": {
      "$controlSwitch": "Information"
    },
    "MinimumLevel": {
      "ControlledBy": "$controlSwitch"
    },
    "WriteTo": [
      {
        "Name": "Seq",
        "Args": {
          "serverUrl": "http://localhost:5341",
          "controlLevelSwitch": "$controlSwitch",
          "apiKey": "DVYuookX2vOq078fuOyJ"
        }
      },
      {
        "Name": "Console",
        "Args": {
          "outputTemplate": "[{Timestamp:HH:mm:ss} {Level}] {SourceContext}{NewLine}{Message:lj}{NewLine}{Exception}{NewLine}",
          "theme": "Serilog.Sinks.SystemConsole.Themes.AnsiConsoleTheme::Literate, Serilog.Sinks.Console"
        }
      }
    ],
    "Enrich": [
      "FromLogContext",
      "WithMachineName",
      "WithThreadId"
    ]
  },
  "InitialData": {
    "Clients": [
      {
        "ClientId": "theidserveradmin",
        "ClientName": "TheIdServer admin SPA Client",
        "ClientUri": "https://localhost:5443/",
        "ClientClaimsPrefix": null,
        "AllowedGrantTypes": [ "authorization_code" ],
        "RequirePkce": true,
        "RequireClientSecret": false,
        "BackChannelLogoutSessionRequired": false,
        "FrontChannelLogoutSessionRequired": false,
        "RedirectUris": [
          "http://localhost:5001/authentication/login-callback",
          "https://localhost:5443/authentication/login-callback"
        ],
        "PostLogoutRedirectUris": [
          "http://localhost:5001/authentication/logout-callback",
          "https://localhost:5443/authentication/logout-callback"
        ],
        "AllowedCorsOrigins": [
          "http://localhost:5001",
          "http://localhost:5002",
          "https://localhost:5443"
        ],
        "AllowedScopes": [
          "openid",
          "profile",
          "theidserveradminapi"
        ],
        "AccessTokenType": "Reference"
      },
      {
        "ClientId": "spa",
        "ClientName": "TheIdServer CUSTOM SPA Client",
        "ClientUri": "https://localhost:5002/",
        "ClientClaimsPrefix": null,
        "AllowedGrantTypes": [ "authorization_code" ],
        "RequirePkce": true,
        "RequireClientSecret": false,
        "BackChannelLogoutSessionRequired": false,
        "FrontChannelLogoutSessionRequired": false,
        "RedirectUris": [
          "http://localhost:5002/authentication/login-callback",
          "https://localhost:5443/authentication/login-callback"
        ],
        "PostLogoutRedirectUris": [
          "http://localhost:5002/authentication/logout-callback",
          "https://localhost:5443/authentication/logout-callback"
        ],
        "AllowedCorsOrigins": [
          "http://localhost:5002",
          "https://localhost:5443"
        ],
        "AllowedScopes": [
          "openid",
          "profile",
          "email",
          "phone",
          "address",
          "theidserveradminapi"
        ],
        "AccessTokenType": "Reference"
      },
      {
        "ClientId": "public-server",
        "ClientName": "Public server Credentials Client",
        "ClientClaimsPrefix": null,
        "AllowedGrantTypes": [ "client_credentials" ],
        "ClientSecrets": [
          {
            "Type": "SharedSecret",
            "Value": "84137599-13d6-469c-9376-9e372dd2c1bd"
          }
        ],
        "Claims": [
          {
            "Type": "role",
            "Value": "Is4-Writer"
          },
          {
            "Type": "role",
            "Value": "Is4-Reader"
          }
        ],
        "BackChannelLogoutSessionRequired": false,
        "FrontChannelLogoutSessionRequired": false,
        "AllowedScopes": [
          "openid",
          "profile",
          "theidserveradminapi"
        ],
        "AccessTokenType": "Reference"
      },
      {
        "ClientId": "theidserver-swagger",
        "ClientName": "TheIdServer Swagger UI",
        "ClientClaimsPrefix": null,
        "AllowedGrantTypes": [ "implicit" ],
        "AllowAccessTokensViaBrowser": true,
        "RequireClientSecret": false,
        "BackChannelLogoutSessionRequired": false,
        "FrontChannelLogoutSessionRequired": false,
        "AllowedScopes": [
          "theidserveradminapi",
          "theidservertokenapi"
        ],
        "AllowedCorsOrigins": [
          "https://localhost:5443"
        ],
        "RedirectUris": [
          "https://localhost:5443/swagger/oauth2-redirect.html"
        ]
      }
    ],
    "RelyingParties": [
      {
        "Id": "defaultSaml1",
        "Description": "The default relying party for SAML 1.0 token.",
        "DigestAlgorithm": "http://www.w3.org/2001/04/xmlenc#sha256",
        "SamlNameIdentifierFormat": "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
        "SignatureAlgorithm": "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256",
        "TokenType": "urn:oasis:names:tc:SAML:1.0:assertion",
        "ClaimMappings": [
          {
            "FromClaimType": "name",
            "ToClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"
          },
          {
            "FromClaimType": "sub",
            "ToClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier"
          },
          {
            "FromClaimType": "email",
            "ToClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
          },
          {
            "FromClaimType": "given_name",
            "ToClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname"
          },
          {
            "FromClaimType": "family_name",
            "ToClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname"
          },
          {
            "FromClaimType": "birthdate",
            "ToClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/dateofbirth"
          },
          {
            "FromClaimType": "website",
            "ToClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/webpage"
          },
          {
            "FromClaimType": "gender",
            "ToClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/gender"
          },
          {
            "FromClaimType": "role",
            "ToClaimType": "http://schemas.microsoft.com/ws/2008/06/identity/claims/role"
          }
        ]
      },
      {
        "Id": "defaultSaml2",
        "Description": "The default relying party for SAML 2.0 token.",
        "DigestAlgorithm": "http://www.w3.org/2001/04/xmlenc#sha256",
        "SamlNameIdentifierFormat": "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
        "SignatureAlgorithm": "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256",
        "TokenType": "urn:oasis:names:tc:SAML:2.0:assertion"
      }
    ],
    "ApiScopes": [
      {
        "Name": "theidserveradminapi",
        "DisplayName": "TheIdServer admin API scope",
        "UserClaims": [
          "name",
          "role"
        ]
      },
      {
        "Name": "theidservertokenapi",
        "DisplayName": "TheIdServer token API scope",
        "UserClaims": [
          "name",
          "role"
        ]
      }
    ],
    "Apis": [
      {
        "Name": "theidserveradminapi",
        "DisplayName": "TheIdServer admin API",
        "UserClaims": [
          "name",
          "role"
        ],
        "ApiSecrets": [
          {
            "Type": "SharedSecret",
            "Value": "5b556f7c-b3bc-4b5b-85ab-45eed0cb962d"
          }
        ],
        "Scopes": [
          "theidserveradminapi",
          "theidservertokenapi"
        ]
      }
    ],
    "Users": [
      {
        "UserName": "alice",
        "Email": "alice@theidserver.com",
        "EmailConfirmed": true,
        "PhoneNumber": "+41766403736",
        "PhoneNumberConfirmed": true,
        "Password": "Pass123$",
        "Roles": [
          "Is4-Writer",
          "Is4-Reader"
        ],
        "Claims": [
          {
            "ClaimType": "name",
            "ClaimValue": "Alice Smith"
          },
          {
            "ClaimType": "given_name",
            "ClaimValue": "Alice"
          },
          {
            "ClaimType": "family_name",
            "ClaimValue": "Smith"
          },
          {
            "ClaimType": "middle_name",
            "ClaimValue": "Alice Smith"
          },
          {
            "ClaimType": "nickname",
            "ClaimValue": "alice"
          },
          {
            "ClaimType": "website",
            "ClaimValue": "http://alice.com"
          },
          {
            "ClaimType": "address",
            "ClaimValue": "{ \"street_address\": \"One Hacker Way\", \"locality\": \"Heidelberg\", \"postal_code\": \"69118\", \"country\": \"Germany\" }"
          },
          {
            "ClaimType": "birthdate",
            "ClaimValue": "1970-01-01"
          },
          {
            "ClaimType": "zoneinfo",
            "ClaimValue": "ch"
          },
          {
            "ClaimType": "gender",
            "ClaimValue": "female"
          },
          {
            "ClaimType": "profile",
            "ClaimValue": "http://alice.com/profile"
          },
          {
            "ClaimType": "locale",
            "ClaimValue": "fr"
          },
          {
            "ClaimType": "picture",
            "ClaimValue": "http://alice.com/picture"
          }
        ]
      }
    ]
  },
  "NegotiateOptions": {
    "Enabled": true
  },
  "OpenTelemetryOptions": {
    "Trace": {
      "Sources": [
        "Duende.IdentityServer.Stores",
        "Duende.IdentityServer.Cache",
        "Duende.IdentityServer.Services",
        "Duende.IdentityServer.Validation"
      ],
      "Service": {
        "Name": "TheIdServer.Duende",
        "Version": "next"
      },
      "ConsoleEnabled": false,
      "Honeycomb": {
        "ServiceName": "TheIdServer.Duende",
        "Dataset": "TheIdServer.Duende"
      }
    },
    "Metrics": {
      "Prometheus": {
        "Protected": true
      }
    }
  },
  "WsFederationOptions": {
    "TokenTypesOffered": [
      { "Uri": "urn:oasis:names:tc:SAML:1.0:assertion" },
      { "Uri": "urn:oasis:names:tc:SAML:2.0:assertion" }
    ],
    "ClaimTypesOffered": [
      {
        "Uri": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name",
        "DisplayName": "Name",
        "Description": "The unique name of the user"
      },
      {
        "Uri": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier",
        "DisplayName": "Name ID",
        "Description": "The SAML name identifier of the user"
      },
      {
        "Uri": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress",
        "DisplayName": "E-Mail Address",
        "Description": "The e-mail address of the user"
      },
      {
        "Uri": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname",
        "DisplayName": "Given Name",
        "Description": "The given name of the user"
      },
      {
        "Uri": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname",
        "DisplayName": "Given Name",
        "Description": "The given name of the user"
      },
      {
        "Uri": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname",
        "DisplayName": "Surname",
        "Description": "The surname of the user"
      },
      {
        "Uri": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/dateofbirth",
        "DisplayName": "Birth date",
        "Description": "The birth date of the user"
      },
      {
        "Uri": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/webpage",
        "DisplayName": "Web page",
        "Description": "The wep page of the user"
      }
    ]
  },
  "Saml2POptions": {
    "X509CertificateValidationMode": "None",
    "ContactPersons": [
      {
        "ContactKind": "Technical",
        "Company": "Aguafrommars",
        "GivenName": "Olivier Lefebvre",
        "SurName": "Aguacongas",
        "EmailAddress": "aguacongas@gmail.com"
      }
    ],
    "RevocationMode": "NoCheck",
    "SignatureAlgorithm": "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256",
    "ValidUntil": 365
  },
  "PasswordHasherOptions": {
    "IterationCount": 600000
  },
  "Argon2PasswordHasherOptions": {
    "Interations": 2,
    "Memory": 67108864
  },
  "BcryptPasswordHasherOptions": {
    "WorkFactor": 11
  },
  "ScryptPasswordHasherOptions": {
    "IterationCount": 131072,
    "BlockSize": 8,
    "ThreadCount": 1
  },
  "UpgradePasswordHasherOptions": {
    "HashPrefixMaps": {
      "0": "Microsoft.AspNetCore.Identity.PasswordHasher",
      "1": "Microsoft.AspNetCore.Identity.PasswordHasher",
      "162": "Aguacongas.TheIdServer.Identity.Argon2PasswordHasher.Argon2PasswordHasher",
      "12": "Aguacongas.TheIdServer.Identity.ScryptPasswordHasher.ScryptPasswordHasher",
      "188": "Aguacongas.TheIdServer.Identity.BcryptPasswordHasher.BcryptPasswordHasher"
    },
    "UsePasswordHasherTypeName": "Aguacongas.TheIdServer.Identity.Argon2PasswordHasher.Argon2PasswordHasher"
  }
}
```

**Reply by KisaPs** [reply]
![image](https://github.com/Aguafrommars/TheIdServer/assets/45015369/94642adb-e427-43f2-951f-68c209c79ee0)

**Reply by aguacongas** [reply]
The really weird thing is the error on Aguacongas.TheIdServer.BlazorApp.Pages.Users.dll because Aguacongas.TheIdServer.BlazorApp.Pages.Users.wasm should be loaded instead of Aguacongas.TheIdServer.BlazorApp.Pages.Users.dll. Did you updated Aguacongas.TheIdServer.BlazorApp.Pages.Users dependency to the latest version ?

And to avoid the refused to loaded browserlink script error you need to update the CSP to allow this script

Your appsettings.json look good BTW
